### PR TITLE
Don't allow userlists to be added to userlists

### LINKED ISSUE #2461: Prevent users from adding lists to lists
Prevent users from adding a user list/path to another userlist/path.

### PR DESCRIPTION
**Pull request by mbertrand**
#### Pre-Flight checklist

- [x] Screenshots and design review for any changes that affect layout or styling
  - [x] Desktop screenshots
  - [ ] Mobile width screenshots
  - [x] Tag @ferdi or @pdpinch for review
- [x] Migrations
  - [x] Migration is backwards-compatible with current production code
- [x] Testing
  - [x] Code is tested
  - [x] Changes have been manually tested

#### What are the relevant tickets?
Closes #2461 

#### What's this PR do?
Prevents users from adding lists to lists

#### How should this be manually tested?
On database migration, any existing userlist items in lists should be removed
Clicking the star on any resource besides a list/path should work the same as before.
Clicking the star on a list/path should only give you "Favorites" as an option

### CODE REVIEW

**Comment by mbertrand** [comment]
@Ferdi @abdulkdawson here's what the 'Add To List' dialog looks like for a list/path: I removed all checkboxes except 'Favorites" as well as the 'Create New List' button - is that okay?

Screen Shot 2019-12-05 at 3 25 41 PM

**Comment by Ferdi** [comment]
@mbertrand I think that's good. Does adding to favorites affects performance ?

**Comment by mbertrand** [comment]
@ferdi yes, a large userlist favorite will have the same problem

File: static/js/components/AddToListDialog.js [hunk 1]
```diff
@@ -123,22 +123,12 @@ export default function AddToListDialog() {
...
-                    item.object_id === userList.id &&
-                    [LR_TYPE_LEARNINGPATH, LR_TYPE_USERLIST].includes(
-                      item.content_type
-                    )
-                )
-              ))) ? (
+      {resource &&
+      ![LR_TYPE_LEARNINGPATH, LR_TYPE_USERLIST].includes(
+        resource.object_type
+      ) ? (
```
Comment (aliceriot) [inline static/js/components/AddToListDialog.js]: we end up making this check in a bunch of places - what do you think about adding a `isUserList` or similar function in `lib/learning_resources`?

Comment (mbertrand) [reply static/js/components/AddToListDialog.js]: Good idea, just pushed that change

**Pull request merged**